src/SFA.DAS.AssessorService.Web.Staff/Validators/Roatp/UpdateApplicationDeterminedDateViewModelValidator.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using FluentValidation;
+using SFA.DAS.AssessorService.Api.Types.Models.Validation;
+using SFA.DAS.AssessorService.Web.Staff.Resources;
+using SFA.DAS.AssessorService.Web.Staff.ViewModels.Roatp;
+
+namespace SFA.DAS.AssessorService.Web.Staff.Validators.Roatp
+{
+    public class UpdateApplicationDeterminedDateViewModelValidator : AbstractValidator<UpdateApplicationDeterminedDateViewModel>
+    {
+        public UpdateApplicationDeterminedDateViewModelValidator()
+        {
+            RuleFor(vm => vm).Custom((vm, context) =>
+            {
+                var validationResult = IsaValidDeterminedDate(vm);
+                if (validationResult.IsValid) return;
+                foreach (var error in validationResult.Errors)
+                {
+                    context.AddFailure(error.Field, error.ErrorMessage);
+                }
+            });
+        }
+
+        private ValidationResponse IsaValidDeterminedDate(UpdateApplicationDeterminedDateViewModel viewModel)
+        {
+            var dateControlName = "ApplicationDeterminedDate";
+
+            var validationResult = new ValidationResponse
+            {
+                Errors = new List<ValidationErrorDetail>()
+            };
+
+            if (viewModel.Day == null && viewModel.Month == null && viewModel.Year == null)
+            {
+                validationResult.Errors.Add(new ValidationErrorDetail(dateControlName, RoatpOrganisationValidation.ApplicationDeterminedDateNoFieldsEntered));
+                validationResult.Errors.Add(new ValidationErrorDetail("Day", "day error"));
```

Comment:
Should these error messages be in a resource file?

Reply:
The words for these are actually not used (they defer to the overacrching control message) but I'll think I'll move them to a resource anyway as a bit of future proofing